wagon-provider-api/src/main/java/org/apache/maven/wagon/authentication/AuthenticationInfo.java

```diff
@@ -51,7 +52,300 @@
+     * used.
+     *
+     *
+     */
+    private String trustStore;
+
+    /**
+     *
+     *
+     * The password to the trust store.
+     *
+     *
+     */
+    private String trustStorePassword;
+
+    /**
+     *
+     *
+     * The type of trust store, default is JKS
+     *
+     * .
+     */
+    private String trustStoreType;
+
+    /**
+     *
+     *
+     * The path to the keystore used for authentication purposes, or null
+     *
+     * .
+     */
+    private String keyStore;
+
+    /**
+     *
+     *
+     * Keystore password, can be null
+     *
+     * .
+     */
+    private String keyStorePassword;
+
+    /**
+     *
+     *
+     * Keystore if the key store has multiple key pairs, this can be used to
+     * explicitly select a specific certificate via it's alias. If null, the
+     * most appropriate certificate is automatically selected by the SSL Factory
+     *
+     * .
+     */
+    private String keyAlias;
+
+    /**
+     *
+     *
+     * The password to unlock the key, can be null
+     *
+     * .
+     */
+    private String keyPassword;
+
+    /**
+     *
+     *
+     * The key store type, defaults to JKS
+     *
+     * .
+     */
+    private String keyStoreType;
+
+    /**
+     *
+     *
+     * The path to the trust store. If not defined, the JRE's cacert store is
+     * used.
+     *
+     *
+     * @return path, name or null
+     */
+    public String getTrustStore()
+    {
+        return trustStore;
+    }
+
+    /**
+     *
+     *
+     * The path to the trust store. If not defined, the JRE's cacert store is
+     * used.
+     *
+     *
+     * @param trustStore path name or null
+     */
+    public void setTrustStore( String trustStore )
+    {
+        this.trustStore = trustStore;
+    }
+
+    /**
+     *
+     *
+     * The password to the trust store.
+     *
+     *
+     * @return password or null
+     */
+    public String getTrustStorePassword()
+    {
+        return trustStorePassword;
+    }
+
+    /**
+     *
+     *
+     * The password to the trust store.
+     *
+     *
+     * @param trustStorePassword password or null
+     */
+    public void setTrustStorePassword( String trustStorePassword ) 
+    {
+        this.trustStorePassword = trustStorePassword;
+    }
+
+    /**
+     *
+     *
+     * The type of trust store, default is JKS
+     *
+     * .
+     *
+     * @return type
+     */
+    public String getTrustStoreType()
+    {
+        return trustStoreType;
+    }
 
+    /**
+     *
+     *
+     * The type of trust store, default is JKS
+     *
+     * .
+     *
+     * @param trustStoreType key store type
+     */
+    public void setTrustStoreType( String trustStoreType )
+    {
+        this.trustStoreType = trustStoreType;
+    }
+
+    /**
+     *
+     *
+     * The path to the keystore used for authentication purposes, or null
+     *
+     * .
+     *
+     * @return path, named keystore (such as MY) or null
```

Comment:
Bad example, if at all `Windows-MY`.

Reply:
ok i'll remove the reference, but windows-my, while it is obviously a platform specific thing, as is the mac specific version, does work and is (in some cases) the only way to access specific certificates